client/dnf_zsync.py

```diff
@@ -36,37 +36,62 @@ def get_input_name(self, repomd, file_name):
         if repomd:
             return re.search(
                 r'<location href=\"repodata/(.*' +
-                self.repodata_base_name(file_name) +
-                r')\"',
+                file_name + r')\"',
                 repomd
             ).group(1)
         else:
             return file_name
 
-    def sync_metadata(self, cache_dir):
-        def iter_repodata(repomd):
-            return re.finditer(
-                r'<location href=\"repodata/(.*)(\.(?:gz))\"',
-                repomd
-            )
+    def download_wget_file(self, file):
+        check_call(["wget", self.mtdt_url + file, "-O", self._cache_dir +
+                    "/repodata/" + file], stdout=DEVNULL, stderr=DEVNULL)
+
+    def save_repomd(self, repomd):
+        with open(self._cache_dir + '/repodata/repomd.xml', 'w') as repomd_f:
+            repomd_f.write(repomd)
 
+    def download_all_files(self, repomd):
```

Comment:
This function is not needed. You can call `download_wget_file()` on `zsync_download + wget_download`.